wal/decoder.go

```diff
@@ -79,6 +80,9 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	}
 
 	recBytes, padBytes := decodeFrameSize(l)
+	if recBytes >= math.MaxInt64-padBytes {
```

Comment:
i think we have a max wal size, also a max key-value pair size. i feel these can be set as a better limit.

Reply:
> i think we have a max wal size, also a max key-value pair size. i feel these can be set as a better limit.

Do we mean define one for max wal entry size? We have request size limit, but I don't see anywhere we define limit? https://github.com/etcd-io/etcd/blob/1166b1f195efae31439c7b3c913b4ef02e7df889/wal/wal.go#L777-L786

Reply:
we have a request limit. also we know that the throughput wont be unlimited. we probably will only be able to batch at most 1000 requests into one WAL entry at most, usually far less... So I guess a 2GB wal entry limit is large enough...

Reply:
Added a check for entry upper limit. Here I assume no project ever allows more than 10 MB raft message entry.